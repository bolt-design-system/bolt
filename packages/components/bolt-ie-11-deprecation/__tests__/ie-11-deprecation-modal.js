import { render, stopServer } from '../../../testing/testing-helpers';

describe('ie 11 deprecation modal', () => {
  let page;

  afterAll(async () => {
    await stopServer();
  });

  test('Basic usage', async () => {
    const results = await render(
      '@bolt-components-ie-11-deprecation/dist/deprecation-modal.html',
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
