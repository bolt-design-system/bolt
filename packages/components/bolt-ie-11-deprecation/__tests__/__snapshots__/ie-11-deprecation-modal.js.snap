// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ie 11 deprecation modal Basic usage 1`] = `
<script>
  parcelRequire=function(e,r,t,n){var i,o="function"==typeof parcelRequire&&parcelRequire,u="function"==typeof require&&require;function f(t,n){if(!r[t]){if(!e[t]){var i="function"==typeof parcelRequire&&parcelRequire;if(!n&&i)return i(t,!0);if(o)return o(t,!0);if(u&&"string"==typeof t)return u(t);var c=new Error("Cannot find module '"+t+"'");throw c.code="MODULE_NOT_FOUND",c}p.resolve=function(r){return e[t][1][r]||r},p.cache={};var l=r[t]=new f.Module(t);e[t][0].call(l.exports,p,l,l.exports,this)}return r[t].exports;function p(e){return f(p.resolve(e))}}f.isParcelRequire=!0,f.Module=function(e){this.id=e,this.bundle=f,this.exports={}},f.modules=e,f.cache=r,f.parent=o,f.register=function(r,t){e[r]=[function(e,r){r.exports=t},{}]};for(var c=0;c
  <t.length;c++)try{f(t[c])}catch(e){i||(i=e)}if(t.length){var l=f(t[t.length-1]);"object"==typeof exports&&"undefined"!=typeof module?module.exports=l:"function"==typeof define&&define.amd?define(function(){return l}):n&&(this[n]=l)}if(parcelRequire=f,i)throw i;return f}({"yoQO":[function(require,module,exports) {
var define;
var e;!function(n){var t;if("function"==typeof e&&e.amd&&(e(n),t=!0),"object"==typeof exports&&(module.exports=n(),t=!0),!t){var o=window.Cookies,r=window.Cookies=n();r.noConflict=function(){return window.Cookies=o,r}}}(function(){function e(){for(var e=0,n={};e
  <arguments.length;e++){var t=arguments[e];for(var o in t)n[o]=t[o]}return n}function n(e){return e.replace(/(%[0-9A-Z]{2})+/g,decodeURIComponent)}return function t(o){function r(){}function i(n,t,i){if("undefined"!=typeof document){"number"==typeof(i=e({path:"/"},r.defaults,i)).expires&&(i.expires=new Date(1*new Date+864e5*i.expires)),i.expires=i.expires?i.expires.toUTCString():"";try{var c=JSON.stringify(t);/^[\\{\\[]/.test(c)&&(t=c)}catch(a){}t=o.write?o.write(t,n):encodeURIComponent(String(t)).replace(/%(23|24|26|2B|3A|3C|3E|3D|2F|3F|40|5B|5D|5E|60|7B|7D|7C)/g,decodeURIComponent),n=encodeURIComponent(String(n)).replace(/%(23|24|26|2B|5E|60|7C)/g,decodeURIComponent).replace(/[\\(\\)]/g,escape);var f="";for(var u in i)i[u]&&(f+="; "+u,!0!==i[u]&&(f+="="+i[u].split(";")[0]));return document.cookie=n+"="+t+f}}function c(e,t){if("undefined"!=typeof document){for(var r={},i=document.cookie?document.cookie.split("; "):[],c=0;c
  <i.length;c++){var f=i[c].split("="),u=f.slice(1).join("=");t||'"'!==u.charAt(0)||(u=u.slice(1,-1));try{var a=n(f[0]);if(u=(o.read||o)(u,a)||n(u),t)try{u=JSON.parse(u)}catch(p){}if(r[a]=u,e===a)break}catch(p){}}return e?r[e]:r}}return r.set=i,r.get=function(e){return c(e,!1)},r.getJSON=function(e){return c(e,!0)},r.remove=function(n,t){i(n,"",e(t,{expires:-1}))},r.defaults={},r.withConverter=t,r}(function(){})});
},{}],"MLWH":[function(require,module,exports) {
"use strict";var e=t(require("js-cookie"));function t(e){return e&&e.__esModule?e:{default:e}}window.NodeList&&!NodeList.prototype.forEach&&(NodeList.prototype.forEach=function(e,t){t=t||window;for(var i=0;i
  <this.length;i++)e.call(t,this[i],i,this)}),window.addEventListener("load",function(){var t=document.querySelectorAll(".js-dismiss-deprecation-notice"),i=document.querySelector(".js-ie-11-deprecation");i&&e.default.get("dismissIEDeprecationNotice")&&i.classList.add("is-hidden"),t&&t.forEach(function(t){t.addEventListener("click",function(t){t.preventDefault(),e.default.set("dismissIEDeprecationNotice",!0,{expires:1}),i.classList.add("is-animating-out"),setTimeout(function(){i.classList.add("is-hidden"),i.classList.remove("is-animating-out")},1e3)})})});
},{"js-cookie":"yoQO"}]},{},["MLWH"], null)
//# sourceMappingURL=/deprecation-modal.062b2056.js.map
</script>
<style>
  @keyframes fadeIn{0%{opacity:0}to{opacity:1}}@keyframes fadeOut{0%{opacity:1}to{opacity:0}}@keyframes slideInModal{0%{transform:translate3d(-50%,200%,0)}to{transform:translate3d(-50%,-50%,0)}}@keyframes slideOutModal{0%{transform:translate3d(-50%,-50%,0)}to{transform:translate3d(-50%,200%,0)}}.c-bolt-ie-11-deprecation,.c-bolt-ie-11-deprecation__close-link--overlay{display:none}.c-bolt-ie-11-deprecation{box-sizing:border-box;visibility:visible;opacity:0;position:fixed;top:50%;left:50%;transform:translate3d(-50%,-200%,0);z-index:2000;width:80vw;max-width:700px;height:auto;color:#000;font-family:Open Sans,Helvetica Neue,sans-serif;border-radius:6px;background-color:#faf4d1;backface-visibility:hidden;transition:all .2s ease;animation:fadeIn .3s ease 1s forwards;animation-name:fadeIn,slideInModal}.c-bolt-ie-11-deprecation *{box-sizing:border-box}.c-bolt-ie-11-deprecation.is-animating-out{opacity:0;transform:translate3d(-50%,200%,0);animation:fadeOut .2s ease 0s forwards;animation-name:fadeOut,slideOutModal}.c-bolt-ie-11-deprecation.is-animating-out+.c-bolt-ie-11-deprecation__close-link--overlay{opacity:0;pointer-events:none;transition:all 0s linear 0s;animation:none}.c-bolt-ie-11-deprecation.is-hidden~.c-bolt-ie-11-deprecation__close-label--overlay,.c-bolt-ie-11-deprecation.is-hidden~.c-bolt-ie-11-deprecation__close-link--overlay,.c-bolt-ie-11-deprecation:target{display:none}@media (-ms-high-contrast:none),screen and (-ms-high-contrast:active){.c-bolt-ie-11-deprecation:not(.is-hidden),.c-bolt-ie-11-deprecation:not(.is-hidden)~.c-bolt-ie-11-deprecation__close-link--overlay{display:block}}.c-bolt-ie-11-deprecation.is-hidden,.c-bolt-ie-11-deprecation__state{display:none}.c-bolt-ie-11-deprecation__state+.c-bolt-ie-11-deprecation__wrapper .c-bolt-ie-11-deprecation__trigger{z-index:1}.c-bolt-ie-11-deprecation__close-link{position:absolute;top:0;right:0;z-index:1;width:100%;height:100%;outline-offset:-4px}.c-bolt-ie-11-deprecation__close-link--overlay{visibility:visible;z-index:1000;background:rgba(6,9,35,.2);transition:all .3s ease;animation:fadeIn .3s ease 1s forwards}.c-bolt-ie-11-deprecation__close-label.c-bolt-ie-11-deprecation__close-label--overlay,.c-bolt-ie-11-deprecation__close-link--overlay{opacity:0;position:fixed;top:0;left:0;width:100%;height:100%}.c-bolt-ie-11-deprecation__close-label.c-bolt-ie-11-deprecation__close-label--overlay{z-index:1001}.c-bolt-ie-11-deprecation__close-text{position:absolute;width:1px;height:1px;overflow:hidden;margin:-1px;padding:0;border:0;clip:rect(0 0 0 0);clip-path:inset(50%);white-space:nowrap}.c-bolt-ie-11-deprecation__state:checked+.c-bolt-ie-11-deprecation .c-bolt-ie-11-deprecation__close-link,.c-bolt-ie-11-deprecation__state:target+.c-bolt-ie-11-deprecation .c-bolt-ie-11-deprecation__close-label{display:block}.c-bolt-ie-11-deprecation__close{display:inline-block;position:absolute;top:0;right:0;line-height:1}.c-bolt-ie-11-deprecation__close-label{display:inline-block;position:relative;z-index:10;width:3rem;height:3rem;cursor:pointer}.c-bolt-ie-11-deprecation__state:checked+.c-bolt-ie-11-deprecation,.c-bolt-ie-11-deprecation__state:target+.c-bolt-ie-11-deprecation{display:none;visibility:hidden;opacity:0;transform:translate3d(0,-200%,0);height:0;padding:0;pointer-events:none;transition:all .2s ease}.c-bolt-ie-11-deprecation__icon{margin-right:1rem}.c-bolt-ie-11-deprecation__close-icon,.c-bolt-ie-11-deprecation__icon{flex-grow:0;flex-shrink:0.5;width:3rem;height:3rem;color:#000}.c-bolt-ie-11-deprecation__close-icon{padding:.75rem}.c-bolt-ie-11-deprecation__wrapper{display:flex;justify-content:center;align-items:center;position:relative;max-width:1400px;margin-right:auto;margin-left:auto;padding:2rem}.c-bolt-ie-11-deprecation__inline-list{display:inline;margin:0 calc(-1rem - -1px) 0 0;padding:0;letter-spacing:-.5rem}.c-bolt-ie-11-deprecation__inline-list-item{display:inline-block;margin:0 .5rem 0 0;padding:0 .5rem 0 0;list-style:none;letter-spacing:normal}.c-bolt-ie-11-deprecation__inline-list-item:not(:last-child){border-right:1px solid rgba(6,9,35,.2)}.c-bolt-ie-11-deprecation__text{display:inline;margin-right:1rem;font-size:14px}.c-bolt-ie-11-deprecation__paragraph{display:inline;margin-right:.5rem}.c-bolt-ie-11-deprecation__link{color:#545da6;text-decoration:underline}.c-bolt-ie-11-deprecation__state:checked~.c-bolt-ie-11-deprecation__close-link,.c-bolt-ie-11-deprecation__state:target~.c-bolt-ie-11-deprecation__close-link--overlay{display:none}
/*# sourceMappingURL=/deprecation-modal.a9f7b6b3.css.map */
</style>
<input class="c-bolt-ie-11-deprecation__state"
       type="checkbox"
       id="close-ie-deprecation"
>
<div class="c-bolt-ie-11-deprecation js-ie-11-deprecation">
  <div class="c-bolt-ie-11-deprecation__wrapper">
    <svg class="c-bolt-ie-11-deprecation__icon"
         xmlns="http://www.w3.org/2000/svg"
         viewbox="0 0 24 24"
    >
      <g fill="none"
         fill-rule="evenodd"
      >
        <g fill="currentColor">
          <path d="M12 21c-4.962 0-9-4.038-9-9s4.038-9 9-9 9 4.038 9 9-4.038 9-9 9m0-20C5.937 1 1 5.938 1 12s4.938 11 11 11 11-4.938 11-11S18.062 1 12 1m0 10c-.55 0-1 .45-1 1v4c0 .55.45 1 1 1s1-.45 1-1v-4c0-.55-.45-1-1-1m0-4a1.001 1.001 0 000 2 1.001 1.001 0 000-2">
          </path>
        </g>
      </g>
    </svg>
    <div class="c-bolt-ie-11-deprecation__text">
      <p class="c-bolt-ie-11-deprecation__paragraph">
        We noticed that your browser is outdated. For the best site experience, please use:
      </p>
      <ul class="c-bolt-ie-11-deprecation__inline-list">
        <li class="c-bolt-ie-11-deprecation__inline-list-item">
          <a href="https://www.google.com/chrome"
             target="_blank"
             class="c-bolt-ie-11-deprecation__link"
          >
            Google Chrome
          </a>
        </li>
        <li class="c-bolt-ie-11-deprecation__inline-list-item">
          <a href="https://www.mozilla.org/firefox"
             target="_blank"
             class="c-bolt-ie-11-deprecation__link"
          >
            Mozilla Firefox
          </a>
        </li>
        <li class="c-bolt-ie-11-deprecation__inline-list-item">
          <a href="https://www.microsoft.com/edge"
             target="_blank"
             class="c-bolt-ie-11-deprecation__link"
          >
            Microsoft Edge
          </a>
        </li>
      </ul>
    </div>
    <div class="c-bolt-ie-11-deprecation__close">
      <a href="#close-ie-deprecation"
         class="c-bolt-ie-11-deprecation__close-link js-dismiss-deprecation-notice"
      >
        <span class="c-bolt-ie-11-deprecation__close-text">
          Close Deprecation Notice
        </span>
      </a>
      <label for="close-ie-deprecation"
             class="c-bolt-ie-11-deprecation__close-label js-dismiss-deprecation-notice"
      >
        <span class="c-bolt-ie-11-deprecation__close-text">
          Close Deprecation Notice
        </span>
        <svg xmlns="http://www.w3.org/2000/svg"
             class="c-bolt-ie-11-deprecation__close-icon"
             viewbox="0 0 24 24"
        >
          <path fill="currentColor"
                fill-rule="evenodd"
                d="M13.414 11.998l5.296-5.294a1.005 1.005 0 000-1.413 1.005 1.005 0 00-1.414 0L12 10.585 6.704 5.291a1.006 1.006 0 00-1.413 0 1.004 1.004 0 000 1.413l5.296 5.294-5.296 5.295a.999.999 0 101.413 1.413l5.295-5.295 5.296 5.295a.999.999 0 001.413 0 1.005 1.005 0 000-1.413l-5.296-5.295h.002z"
          >
          </path>
        </svg>
      </label>
    </div>
  </div>
</div>
<div class="c-bolt-ie-11-deprecation__close-link c-bolt-ie-11-deprecation__close-link--overlay">
</div>
`;
